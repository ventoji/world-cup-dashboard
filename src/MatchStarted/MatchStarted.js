import React, { useState } from 'react';
import PropTypes from 'prop-types';
import InputWC from '../InputWC';
import ButtonWC from '../ButtonWC';
import { validateScoreWCfields } from '../utils/validators';


const SCORE_STYLE = {
    li: {
      display: 'flex',
      alignContent: 'space-around',
      marginBottom: '10px',
    },
  };

const MatchStarted = ({ match, updateScore, handleFinish }) => {
    const [matchScore, setMatchScore] = useState({
      startAt: match.startAt,
      id: match.id,
      homeTeam: match.homeTeam,
      awayTeam: match.awayTeam,
      homeScore: 0,
      awayScore: 0,
    });
    const [errors, setErrors] = useState({});
  
    const handleMatchScore = (event) => {
  
    const field = event.target.name;
    const value = event.target.value;
    const errors = validateScoreWCfields(field,value);
    setErrors(errors);

      setMatchScore({
        ...matchScore,
        [field]: value >= 0 ? value : matchScore[field]
      });
   
    };
  
    const submitScore = (event) => {
      event.preventDefault();
      updateScore({
        id: matchScore.id,
        homeScore: matchScore.homeScore,
        awayScore: matchScore.awayScore,
      });
    };
  
    const submitFinishGame = (event) => {
      event.preventDefault();
      handleFinish(matchScore);
    };
  
    return (
      <li className="list-player" style={SCORE_STYLE.li}>
        <span style={{ color: 'blue' }}> Match: </span>{' '}
        <InputWC
          htmlId="homeScore"
          label={match.homeTeam}
          value={matchScore.homeScore}
          name="homeScore"
          type="number"
          style={{ width: '35%' }}
          onChange={handleMatchScore}
          error={errors.homeScore}
        />{' '}
        -
        <InputWC
          htmlId="awayScore"
          label={match.awayTeam}
          value={matchScore.awayScore}
          name="awayScore"
          type="number"
          style={{ width: '35%' }}
          onChange={handleMatchScore}
          error={errors.awayScore}
        />
        <ButtonWC
          label="update"
          style={{ height: '35%' }}
          onClick={submitScore}
        />
        <ButtonWC
          label="finish"
          style={{ height: '35%' }}
          onClick={submitFinishGame}
        />
      </li>
    );
  };

  MatchStarted.propTypes = {
    /** Contains info about players and scores */
    match: PropTypes.object,
     /** Function to update Score */
    updateScore: PropTypes.func.isRequired,
      /** Function to finish a game */
    handleFinish: PropTypes.func.isRequired
  };
  
  MatchStarted.defaultProps = {
    match: {},
    updateScore: () => {},
    finishGame: () => {}
  };
  
  export default MatchStarted;