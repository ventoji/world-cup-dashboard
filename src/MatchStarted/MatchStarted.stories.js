import React from 'react';
import MatchStarted from './MatchStarted';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs';

export default {
  title: 'WC Components/MatchStarted',
  component: MatchStarted,
  decorators: [withKnobs]
};

const match =  {
  id: 1,
  startAt: 1616113389731,
  homeTeam: 'Mexico',
  awayTeam: 'Canada',
  homeScore: 0,
  awayScore: 0,
};

const Template = (args) => <MatchStarted {...args} />;

export const Default = Template.bind({});

Default.args = {
  match: match,
  updateScore: action('cick'),
  handleFinish: action('cick')
};