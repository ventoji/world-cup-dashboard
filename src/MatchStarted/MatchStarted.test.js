import React from 'react';
import MatchStarted from './MatchStarted';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('MatchStarted  Component', () => {
  let matchStarted;
  const match = {
    id: 1,
    startAt: 1616113389731,
    homeTeam: 'Mexico',
    awayTeam: 'Canada',
    homeScore: 0,
    awayScore: 0,
  };

  beforeEach(() => {
    matchStarted = shallow(
      <MatchStarted
        match={match}
        updateScore={() => {}}
        handleFinish={() => {}}
      />,
    );
  });

  it('Should be defined', () => {
    expect(MatchStarted).toBeDefined();
  });

  it('Should render properly', () => {
    let element = matchStarted.find('InputWC');

    expect(toJson(matchStarted)).toMatchSnapshot();
    expect(element).toHaveLength(2);

    element = matchStarted.find('ButtonWC');
    expect(element).toHaveLength(2);
  });
});
