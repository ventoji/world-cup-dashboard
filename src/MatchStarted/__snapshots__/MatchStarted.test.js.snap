// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchStarted  Component Should render properly 1`] = `
<li
  className="list-player"
  style={
    Object {
      "alignContent": "space-around",
      "display": "flex",
      "marginBottom": "10px",
    }
  }
>
  <span
    style={
      Object {
        "color": "blue",
      }
    }
  >
     Match: 
  </span>
   
  <InputWC
    htmlId="homeScore"
    label="Mexico"
    name="homeScore"
    onChange={[Function]}
    placeholder="value here"
    required={true}
    style={
      Object {
        "width": "35%",
      }
    }
    type="number"
    value={0}
  />
   
  -
  <InputWC
    htmlId="awayScore"
    label="Canada"
    name="awayScore"
    onChange={[Function]}
    placeholder="value here"
    required={true}
    style={
      Object {
        "width": "35%",
      }
    }
    type="number"
    value={0}
  />
  <ButtonWC
    label="update"
    onClick={[Function]}
    style={
      Object {
        "height": "35%",
      }
    }
  />
  <ButtonWC
    label="finish"
    onClick={[Function]}
    style={
      Object {
        "height": "35%",
      }
    }
  />
</li>
`;
